### PR TITLE
[8.5][DOCS] Adds ML-related highlights to 8.5

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds ML-related release highlights text to the yaml file that has been generated when the unattended transform mode was introduced. It adds two items to the highlights section; the first one is about the unattended transform mode, and the second one is about the frequent items aggregation.

### CODE REVIEW

File: docs/changelog/89212.yaml [hunk 1]
```diff
@@ -3,3 +3,27 @@ summary: Add an unattended mode setting to transform
...
+    require direct intervention to resolve. For example, a high disk watermark 
+    may prevent all ingest activities, including the indexing operation of a 
+    transform; however once the original issue is resolved by the system 
+    administrator, the transform automatically continue to operate from where it 
+    stopped working. Neither the system administrator nor the end-user need to 
+    restart the transform.
+
+    [discrete]
+    [[frequent-items-agg]]
+    === Frequent items aggregation
```
Comment (droberts195) [inline docs/changelog/89212.yaml:20]: Is the way this has been done going to mean the frequent items aggregation shows up as a transform feature?

Comment (szabosteve) [reply docs/changelog/89212.yaml:20]: @droberts195 
It won't show up as a transform feature. 
The items in the PR will show up as two distinct sections on the What's new page. For example, in the [8.4 highlights](https://www.elastic.co/guide/en/elasticsearch/reference/current/release-highlights.html#composite_aggregations_in_datafeeds_are_generally_available), the sections about composite aggs and the optimized DFA speed was added in the same PR (similar to this situation). The only indicator that the two items were added in the same PR is the pull request reference after the second section. The workflow is far from ideal, we have an [issue](https://github.com/elastic/platform-docs-team/issues/38) for improving the process, so hopefully, we don't need to publish the release highlights in this "hacky" way in the future.

File: docs/changelog/89212.yaml [hunk 2]
```diff
@@ -3,3 +3,27 @@ summary: Add an unattended mode setting to transform
...
 issues: []
+highlight:
+  title: Unattended mode for transforms
+  body: |-
+    The new `unattended` setting for transforms introduces unattended mode. It 
+    enables the transform to retry indefinitely, even in the face of errors that 
+    require direct intervention to resolve. For example, a high disk watermark 
+    may prevent all ingest activities, including the indexing operation of a 
+    transform; however once the original issue is resolved by the system 
+    administrator, the transform automatically continue to operate from where it
```
Comment (droberts195) [inline docs/changelog/89212.yaml:14]: There's something missing.  Could be "the transform will automatically" or "automatically continues" - whatever matches the house style best.

File: docs/changelog/89212.yaml [hunk 3]
```diff
@@ -3,3 +3,27 @@ summary: Add an unattended mode setting to transform
...
+    [discrete]
+    [[frequent-items-agg]]
+    === Frequent items aggregation
+
+    The 
+    {ref}/search-aggregations-bucket-frequent-items-aggregation.html[frequent items aggregation] 
+    is a new bucket aggregation which identifies items that often occur 
+    together. It is a form of association rules mining that helps to discover 
+    relationships between different data points. For example, the aggregation 
+    can find log events that tend to co-occur and may give a more informative
```
Comment (droberts195) [inline docs/changelog/89212.yaml:27]:
```suggestion
    can find attributes of log events that tend to co-occur and may give a more informative 
```

File: docs/changelog/89212.yaml [hunk 4]
```diff
@@ -3,3 +3,27 @@ summary: Add an unattended mode setting to transform
...
+    restart the transform.
+
+    [discrete]
+    [[frequent-items-agg]]
+    === Frequent items aggregation
+
+    The 
+    {ref}/search-aggregations-bucket-frequent-items-aggregation.html[frequent items aggregation] 
+    is a new bucket aggregation which identifies items that often occur 
+    together. It is a form of association rules mining that helps to discover
```
Comment (droberts195) [inline docs/changelog/89212.yaml:25]:
```suggestion
    together. It is a form of association rule mining that helps to discover 
```

File: docs/changelog/89212.yaml (hunk 2 above)
Comment (lcawl) [inline docs/changelog/89212.yaml:14]:
```suggestion
    administrator, the transform automatically continues to operate from where it 
```

**Review by lcawl — APPROVED** [review]
[Preview](https://elasticsearch_90506.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/8.5/release-highlights.html) LGTM

**Review by droberts195 — APPROVED** [review]
LGTM2

**Pull request merged**